plugin/inlay_hint.py

```diff
@@ -140,13 +141,21 @@ def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uui
                 }
             })
             result += f'<a href="{inlay_hint_click_command}">'
+        truncated = len(label) > truncate_limit and truncate_limit > 0
+        truncated_label = label[:truncate_limit - 1] + '…' if truncated else label
         instruction_text = '\nDouble-click to insert' if has_text_edits else ""
-        result += f'<span title="{(tooltip + instruction_text).strip()}">{html.escape(label)}</span>'
+        truncation_tooltip = f'\n{html.escape(label)}' if truncated else ""
+        result_tooltip = (tooltip + instruction_text + truncation_tooltip).strip()
+        result += f'<span title="{result_tooltip}">{html.escape(truncated_label)}</span>'
         if is_clickable:
             result += "</a>"
         return result
-
+    remaining_truncate_limit = truncate_limit
+    full_label = "".join(label_part['value'] for label_part in label)
+    full_label_truncated = len(full_label) >= truncate_limit and truncate_limit > 0
```

Comment:
Should this also be `>` here or not?

---

I have tested the PR with a server that uses the regular label (no label parts) and I think it works well with the tooltips.

My only suggestion left would be to use a lower default value for the setting. I think 30 would be a sane default. I would say the truncation is actually a good thing to do, and some users might not check settings or read update notes, so it should be applied by default. Opinions?

Reply:
> Should this also be `>` here or not?
> 
> I have tested the PR with a server that uses the regular label (no label parts) and I think it works well with the tooltips.
> 
> My only suggestion left would be to use a lower default value for the setting. I think 30 would be a sane default. I would say the truncation is actually a good thing to do, and some users might not check settings or read update notes, so it should be applied by default. Opinions?

I'll check it out. Thanks for pointing that out.

---

I think inlay hint is not enabled by default, so most users will not be affected. So I agree that this feature should be enabled by default. I'll wait for more opinions.

Reply:
> I'll check it out. Thanks for pointing that out.

Done in [d6a544a](https://github.com/sublimelsp/LSP/pull/2514/commits/d6a544a1fa3a558b16bb485fd454fed6a07d5777).

---

Done in [a43e68a](https://github.com/sublimelsp/LSP/pull/2514/commits/a43e68adbd1c500e8f887f626bf5282e6a72b11e)

I will not resolve this conversation in case of reconsideration, should I?